mm2src/mm2_main/src/lp_swap/taker_restart.rs

```diff
@@ -49,6 +49,7 @@ pub async fn get_command_based_on_maker_or_watcher_activity(
             Err(e) => ERR!("Error {} when trying to find taker payment spend", e),
         },
         TakerSwapCommand::SpendMakerPayment => check_maker_payment_spend_and_add_event(ctx, swap, saved).await,
+        TakerSwapCommand::ConfirmMakerPaymentSpend => Ok(command),
```

Comment:
The below is related to this change in `get_command_based_on_maker_or_watcher_activity`.

I think we should return `TakerSwapCommand::ConfirmMakerPaymentSpend` instead of `TakerSwapCommand::Finish` here https://github.com/KomodoPlatform/komodo-defi-framework/blob/1c3c935d8fa4581603f48ea031722b90ca8d3b7a/mm2src/mm2_main/src/lp_swap/taker_restart.rs#L145

Reply:
c.c. @dimxy

Reply:
Seems so.
search_for_swap_tx_spend_other fn may return unconfirmed txns.
So we need (with the created event MakerPaymentSpentByWatcher) to go to ConfirmMakerPaymentSpend command.
I think it's better to fix also get_command() to ensure MakerPaymentSpentByWatcher event triggers ConfirmMakerPaymentSpend.
And good to have a new swap watcher test for this to ensure the flow works okay

Reply:
> I think it's better to fix also get_command() to ensure MakerPaymentSpentByWatcher event triggers ConfirmMakerPaymentSpend.

yep, if we want to return `TakerSwapCommand::ConfirmMakerPaymentSpend` in `check_maker_payment_spend_and_add_event`, then we also need to cover confirmation step in `get_command` for `MakerPaymentSpentByWatcher`

Reply:
> > I think it's better to fix also get_command() to ensure MakerPaymentSpentByWatcher event triggers ConfirmMakerPaymentSpend.
> 
> yep, if we want to return `TakerSwapCommand::ConfirmMakerPaymentSpend` in `check_maker_payment_spend_and_add_event`, then we also need to cover confirmation step in `get_command` for `MakerPaymentSpentByWatcher`

Hmm it means that we also need to update `fn is_recoverable` according to the same note as here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199#discussion_r1776229321

Reply:
@shamardy one moment about watchers, do we actually use this feature in production? Im not sure that it is fully functional in prod.

If nope, we shouldn't change `MakerPaymentSpentByWatcher` handle in legacy `get_command()`function

Reply:
We might need to handle it here too https://github.com/KomodoPlatform/komodo-defi-framework/blob/aed8d4c50fe357f0d23529af0e71d0e3091c40ea/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L187

Reply:
> It's used for utxo <-> utxo swap in production. 

Got it, then I also update MakerPaymentSpentByWatcher cases

Reply:
> We might need to handle it here too
> 
> https://github.com/KomodoPlatform/komodo-defi-framework/blob/aed8d4c50fe357f0d23529af0e71d0e3091c40ea/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L187

yep, its what dimxy was referencing about

Reply:
Cover `TakerSwapCommand::ConfirmMakerPaymentSpend` for restart and watchers here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199/commits/0661ee22ab82ed6f09de110f0cadb902746ea39a

> And good to have a new swap watcher test for this to ensure the flow works okay

@dimxy new events matching in `get_command` function already covered in watcher (restart) tests in `swap_watchers_test.rs` , comparing expected event list with actual using `check_actual_events` function
https://github.com/KomodoPlatform/komodo-defi-framework/blob/0661ee22ab82ed6f09de110f0cadb902746ea39a/mm2src/mm2_main/tests/docker_tests/swap_watcher_tests.rs#L264

for example I removed `"MakerPaymentSpendConfirmed"` event from `expected_events` list  in `test_taker_saves_the_swap_as_successful_after_restart_panic_at_maker_payment_spend` test

and got https://github.com/laruh/atomicDEX-API/actions/runs/11091593379/job/30815533174#step:6:3409
```
thread 'docker_tests::swap_watcher_tests::test_taker_saves_the_swap_as_successful_after_restart_panic_at_maker_payment_spend' panicked at 'assertion failed: `(left == right)`
  left: `["Started", "Negotiated", "TakerFeeSent", "TakerPaymentInstructionsReceived", "MakerPaymentReceived", "MakerPaymentWaitConfirmStarted", "MakerPaymentValidatedAndConfirmed", "TakerPaymentSent", "WatcherMessageSent", "TakerPaymentSpent", "MakerPaymentSpentByWatcher", "Finished"]`,
 right: `["Started", "Negotiated", "TakerFeeSent", "TakerPaymentInstructionsReceived", "MakerPaymentReceived", "MakerPaymentWaitConfirmStarted", "MakerPaymentValidatedAndConfirmed", "TakerPaymentSent", "WatcherMessageSent", "TakerPaymentSpent", "MakerPaymentSpentByWatcher", "MakerPaymentSpendConfirmed", "Finished"]`', mm2src/mm2_main/tests/docker_tests/swap_watcher_tests.rs:269:5
```

Reply:
Check if this is resolved @dimxy , it looks good to me.